docs/get_started/debugging.md

```diff
@@ -1,2 +1,17 @@
 # Debugging
-For compiling in debug mode replace `cmake ..` with `cmake -DCMAKE_BUILD_TYPE=Debug ..` (this will allow to step through with a debugger).
\ No newline at end of file
+When debugging, some additional compiler flags are needed to produce debugging information. These flags can be added using the CMake's [`CMAKE_BUILD_TYPE`](https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html) environment variable by:
+
+1. specifying `-DCMAKE_BUILD_TYPE=Debug` when executing `cmake` in the command line,
+2. putting `set(CMAKE_BUILD_TYPE "Debug")` into *yours* or the *libtropic's* main `CMakeLists.txt`.
+
+After this, you can use debugging tools of your choice, e.g. [gdb](https://www.gnu.org/savannah-checkouts/gnu/gdb/index.html) or [Valgrind](https://valgrind.org/), which are available straight from the command line by writing `gdb` or `valgrind` before the name of the binary.
```

Comment:
"which are available straight from the command line" --> they are not, if the user did not install them :D I would probably leave only the first sentence or add sth like "refer to their doc":

```suggestion
After this, you can use debugging tools of your choice, e.g. [gdb](https://www.gnu.org/savannah-checkouts/gnu/gdb/index.html) or [Valgrind](https://valgrind.org/). Refer to each tool's documentation for the installation instructions.
```